File: DONGUK/프로그래머스_WEEK 5/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 0;

        while(a != b){
            answer++;
            
            if(a%2 == 1){
                a += 1;
            }
            
            if(b%2 == 1){
                b += 1;
            }
            
            a = a / 2;
            b = b / 2;
        }
        return answer;
    }
}

File: JUNHEE/프로그래머스_week5/예상 대진표.cpp
#include <iostream>

using namespace std;

int solution(int n, int a, int b)
{
    int answer = 0;

    while(true){
        answer++;
        if(abs(a-b) == 1 && ((a > b && a%2 == 0) || (b > a && b % 2 == 0)))
            break;
        
        a = (a + 1) / 2;
        b = (b + 1) / 2;
    }

    return answer;
}
